controllers/statusreport/statusreport_controller.go

```diff
@@ -81,8 +81,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	adapter := NewAdapter(snapshot, application, logger, loader, r.Client, ctx)
 	return controller.ReconcileHandler([]controller.Operation{
-		adapter.EnsureSnapshotTestStatusReportedToGitHub,
 		adapter.EnsureSnapshotFinishedAllTests,
+		adapter.EnsureSnapshotTestStatusReportedToGitHub,
```

Comment:
Should be also diagram updated?

Reply:
What is the reason exchange the order of them? Run `EnsureSnapshotFinishedAllTests` firstly and then `EnsureSnapshotTestStatusReportedToGitHub`? But `EnsureSnapshotFinishedAllTests` need the test status written by `EnsureSnapshotTestStatusReportedToGitHub`, is my understanding wrong?

Reply:
> What is the reason exchange the order of them?

Before the order was reversed:
* the finalizer from Integration PLRs
  (related to the pull_request event) is not removed,
  even after the Snapshot gets marked as Passed/Failed.
* As we know, the EnsureSnapshotFinishedAllTests function
  marks the Snapshot as Passed/Failed.
* And since the EnsureSnapshotFinishedAllTests function
  is called AFTER EnsureSnapshotTestStatusReportedToGitHub
  function, the finalizer is never removed.

After the order was reversed:
* The finalizer gets removed successfully.

> But EnsureSnapshotFinishedAllTests need the test status written by EnsureSnapshotTestStatusReportedToGitHub

As per my understanding, the `EnsureSnapshotFinishedAllTests` function needs the latest test status data from the Snapshot's annotation. And `EnsureSnapshotTestStatusReportedToGitHub` function doesn't make any changes to that annotation, so there's no dependency between them in that way.

Reply:
Yes, you are right. I am wrong. Thanks!